core/server/inventory.go

```diff
@@ -145,16 +142,15 @@ func (cs *coreServer) getHelmReleaseInventory(ctx context.Context, k8sClient cli
 
 // Returns the list of resources applied in the helm chart.
 func getHelmReleaseObjects(ctx context.Context, k8sClient client.Client, helmRelease *helmv2.HelmRelease) ([]*unstructured.Unstructured, error) {
-	storageNamespace := helmRelease.GetStorageNamespace()
-
-	storageName := helmRelease.GetReleaseName()
-
-	storageVersion := helmRelease.Status.LastReleaseRevision
+	storageVersion := storageVersionFromHelmRelease(helmRelease)
 	if storageVersion < 1 {
 		// skip release if it failed to install
 		return nil, nil
 	}
 
+	storageNamespace := helmRelease.GetStorageNamespace()
+	storageName := helmRelease.GetReleaseName()
```

Comment:
It would be better to collect this from the `Current` as well, which contains a `.Name`. Something like:

```go
func storageDataFromHelmRelease(helmRelease *helmv2.HelmRelease) (string, int) {
	if latest := helmRelease.Status.History.Latest(); latest != nil {
		return latest.Name, latest.Version
	}
	return helmRelease.GetReleaseName(), helmRelease.Status.LastReleaseRevision
}
```

Reply:
Acually, if you do this. It would simply be better to also collect the `.Status.StorageNamespace`. So you would end up with something like `(ns string, name string, version int)`, and:

```
helmrelease.Status.StorageNamespace, latest.Name, latest.Version
```

or

```
helmRelease.GetStorageNamespace(), helmRelease.GetReleaseName(), helmRelease.Status.LastReleaseRevision
```